libs/core/core/hooks/use-msc-device-detached-effect.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { useCallback, useEffect } from "react"
+import { useDispatch, useSelector } from "react-redux"
+import { answerMain, useDebouncedEventsHandler } from "shared/utils"
+import {
+  DeviceBaseProperties,
+  DeviceProtocolMainEvent,
+  DeviceType,
+} from "device-protocol/models"
+import {
+  abortMscFlashing,
+  FlashingProcessState,
+  selectIsFlashingInActivePhases,
+} from "msc-flash-harmony"
+import { Dispatch } from "Core/__deprecated__/renderer/store"
+
+export const useMscDeviceDetachedEffect = () => {
```

Comment:
Why another hook if we already have the `useDeviceDetachedEffect` ?

Reply:
Initially, I wanted to handle this code in a single hook, but due to circular dependencies, I had to add a new hook related to detach. I suggest leaving it as is and creating a technical task where we can merge the three hooks into one and resolve the dependency issue.

Reply:
In case of dependencies issues you can try an import like "../../../../../../../........"

Reply:
https://appnroll.atlassian.net/browse/CP-3194